server/service/live_queries.go

```diff
@@ -278,9 +278,17 @@ func (svc *Service) RunLiveQueryDeadline(
 			defer cancelFunc()
 
 			defer func() {
+				// We do not want to use the outer `ctx` because we want to cleanup the campaign
+				// even if such `ctx` is canceled (e.g17197. a client terminating the connection).
+				ctx := context.Background()
```

Comment:
Nit: you should probably use `context.WithoutCancel` (https://pkg.go.dev/context#WithoutCancel) so that whatever tracing/logging stuff we have in the context is still there, it just removes the cancellation part (relatively new but we do use Go 1.21+ already).